Assets/Scripts/Game Singal/GameSignal.cs

```diff
@@ -0,0 +1,29 @@
+﻿#pragma warning disable 0649
+
+using System.Collections.Generic;
+using UnityEngine;
+
+[CreateAssetMenu]
+public class GameSignal : ScriptableObject
+{
+
+    [SerializeField] private List<IGameSignalListener> listeners = new List<IGameSignalListener>();
+
+    public void Raise(float value)
+    {
+        for (int i = listeners.Count - 1; i >= 0; i--)
+        {
+            listeners[i].OnGameSignalRaised(value);
```

Comment:
This is pretty cool, we can use it for all sorts of stuff related to the GUI.

I would recommend adding a 'channel' or 'EventType' enum or some similar concept though, to ensure that there are not collisions between one signal and another - so for example, we can set a float for concentration in the UI without setting all of our other listeners.